util/merkle/benches/triedb.rs

```diff
@@ -92,14 +92,14 @@ impl TestDB {
         }
     }
 
-    fn trie(&self) -> TrieDB {
+    fn trie<'db>(&'db self) -> Box<dyn Trie + 'db> {
```

Comment:
The return type became a lot verbose than before, and it seems it's coming from the implementation of `TrieFactory`'s methods.
I suggest simplifying this by refactoring `TrieFactory` and remove those `Box` and lifetimes if possible.

Reply:
You cannot remove Box because the size of the return type should be known at the compile time.

Reply:
Oh, I see.

Reply:
Ah, I think `impl Trait` can work. I'll try it.

Reply:
`impl Trait` worked.